### PR TITLE
miniscript: init at unstable-20201104

### PR DESCRIPTION
**Pull request by RaghavSood**
###### Motivation for this change

Miniscript is a tool to compile and work with the miniscript Bitcoin policy language.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/miniscript/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "miniscript";
+  version = "unstable-20201104";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/miniscript/default.nix:5]:
```suggestion
  version = "unstable-2020-11-04";
```

File: pkgs/applications/blockchains/miniscript/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sipa";
+    repo = pname;
+    rev = "5115dd0bdf70c3e3b728c2c0e9c8af4b7edd13ed";
+    sha256 = "0bklhlnl4wkb90rpqhs5n0h6rmnyb3js8ldgynppn27b2kbbmik5";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/miniscript/default.nix:15]: You should execute the pre and post install hooks.

File: pkgs/applications/blockchains/miniscript/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    cp miniscript $out/bin/miniscript
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Miniscript is a language for writing (a subset of) Bitcoin Scripts in a structured way, enabling analysis, composition, generic signing and more.";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/miniscript/default.nix:20]: This is description is quite long. It should probably be moved to ``longDescription`` and a shorter one for the short should be used.

**Comment by RaghavSood** [comment]
@SuperSandro2000 appreciate the quick review, updated

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102808` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>miniscript</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Fails to build on darwin with:

```
building
build flags: SHELL=/nix/store/7kzamk0abm530kkr5v106hnc3cfscwkc-bash-4.4-p23/bin/bash
g++ -O3 -g0 -Wall -std=c++11 -march=native -flto -Ibitcoin bitcoin/util/strencodings.cpp bitcoin/util/spanparsing.cpp bitcoin/script/script.cpp bitcoin/script/miniscript.cpp compiler.cpp main.cpp -o miniscript
/nix/store/7kzamk0abm530kkr5v106hnc3cfscwkc-bash-4.4-p23/bin/bash: g++: command not found
make: *** [Makefile:5: miniscript] Error 127
```

File: pkgs/applications/blockchains/miniscript/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp miniscript $out/bin/miniscript
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/miniscript/default.nix:21]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
also please mark it broken on darwin.

**Pull request closed without merging**